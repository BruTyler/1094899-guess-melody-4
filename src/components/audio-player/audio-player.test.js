import React from 'react';
import renderer from 'react-test-renderer';
import AudioPlayer from './audio-player.jsx';

const CLICKHANDLER = () => {};
const QUESTION_ARTIST = {
  song: {
    src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`
  }
};

describe(`<AudioPlayer /> render suit`, () => {
  it(`<AudioPlayer /> render case`, () => {
    const generatedTree = renderer.create(
        <AudioPlayer
          src={QUESTION_ARTIST.song.src}
          isPlaying={true}
          isLoading={false}
          onPlayButtonClick={CLICKHANDLER}
        >
          <audio />
        </AudioPlayer>,
        {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

    expect(generatedTree).toMatchSnapshot();
  });
});
