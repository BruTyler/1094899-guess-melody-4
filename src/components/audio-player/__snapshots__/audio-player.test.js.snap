// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioPlayer /> render suit <AudioPlayer /> render case 1`] = `
Array [
  <button
    className="track__button track__button--pause"
    disabled={false}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
